InnerEye/ML/configs/histo_configs/classification/BaseMIL.py

```diff
@@ -44,8 +44,12 @@ class BaseMIL(LightningContainer):
     cache_mode: CacheMode = param.ClassSelector(default=CacheMode.MEMORY, class_=CacheMode,
                                                 doc="The type of caching to perform: "
                                                     "'memory' (default), 'disk', or 'none'.")
-    save_precache: bool = param.Boolean(True, doc="Whether to pre-cache the entire transformed "
-                                                  "dataset upfront and save it to disk.")
+    precache_location: str = param.ClassSelector(default=CacheLocation.NONE, class_=CacheLocation,
+                                                 doc="Whether to pre-cache the entire transformed dataset upfront "
+                                                 "and save it to disk and if re-load in cpu or gpu. Options:"
+                                                 "`none` (default),`cpu`, `gpu`")
```

Comment:
see my comment above, maybe make that optional, rather than encoding None in the enum?

Reply:
I was trying to maintain the same pattern we already had for CacheMode but I understand is not beautiful

Reply:
I left this untouched but removed the horrible construct you commented on above